lib/src/proxy-authenticate.class.ts

```diff
@@ -0,0 +1,48 @@
+import { BaseHeader } from "./base-header.class";
+export class ProxyAuthenticate extends BaseHeader {
+  protected _type?: string | undefined;
+  protected _realm?: string | undefined;
+
+  getType(): string | undefined {
+    return this._type;
+  }
+
+  setType(value: string | undefined) {
```

Comment:
Can we add more specific types here? They are listed on this page: https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes

Can probably make a "[Authentication schemes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Authentication#authentication_schemes)" type

```typescript
type AuthenticationType = 'Basic' | 'Bearer' | etc...
```

```suggestion
  setType(value: AuthenticationType | undefined) {
```

Reply:
Okay I will check and modify as specific types